GLFW fixes

GameParrot opened the pull request:
- ~~macOS fullscreen glitch fixed~~
- DEL key in text boxes fixed
- GAMEWINDOW_CENTER_CURSOR env var
- raw mouse input if supported

~~Fullscreen glitch demo (flashing light warning):~~

~~[fsbug.mp4](https://user-images.githubusercontent.com/85067619/233781498-864ee921-72d1-49c4-8533-dc53bb6b8156.mp4)~~

GameParrot committed:
GLFW fixes

- macOS fullscreen glitch fixed
- DEL key in text boxes fixed

ChristopherHX commented:
Do you see this glitch also in https://github.com/minecraft-linux/game-window/pull/15?
Entering full screen should
- call the scale change callback
- call the window size changed callback
If one of them is called during event polling it makes sense to call onWindowSizeChanged.

I still don't like to call `onWindowSizeChanged(width, height);` with outdated width and height.

ChristopherHX commented:
I can reproduce a completly unrelated fullscreen glitch.

- open the game
- Enter fullscreen
- Leave fullscreen
- change the screen resolution
- Enter fullscreen
- screen resolution is changed by glfw
- fullscreen window surface is floating like a window in the middle of the screen over other window content, which is even interactable.

GameParrot committed:
Fix mosue reset to center glitch

- Mouse no longer resets to center when leaving world
- Mouse should no longer reset to center when navigating menus on some servers (hive treasure wars)

GameParrot committed:
Env var for center, use raw input if supported

GameParrot committed:
remove oms,omx

GameParrot committed:
remove unused code

GameParrot committed:
Merge branch 'minecraft-linux:master' into glfw-fixes

GameParrot committed:
GAMEWINDOW_CENTER_CURSOR for eglut

ChristopherHX commented on src/window_glfw.cpp:
```diff
@@ -158,6 +165,7 @@ void GLFWGameWindow::setFullscreen(bool fullscreen) {
     } else {
         glfwSetWindowMonitor(window, nullptr, windowedX, windowedY, windowedWidth, windowedHeight, GLFW_DONT_CARE);
     }
+    onWindowSizeChanged(width, height);
```
Please remove, I'm not going to merge this specfic change.

GameParrot replied:
removed, that was left by mistake

ChristopherHX reviewed (commented):
One specifc change stops me from merging this.
That's the price you pay for creating a PR with unrelated changes.

GameParrot committed:
Update window_glfw.cpp
